src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
+          debug('not inserted, history was just created or action === redux-undo/INIT')
+        } else {
           history = insert(history, res, config.limit)
           debug('inserted new state into history')
-        } else {
-          debug('not inserted, history is unchanged')
```

Comment:
So yeah.. I reintroduced the INIT action here. Cause we never want to insert when that is fired right? 
And we also don't want to insert when we just created the history (Redux init) 

All other actions will be recorded again, so also the actions resulting in the same state as before. As long as the distinctState filter is not active. I think that is the preferred behavior.